server/test/write_authzmodel.go

```diff
@@ -7,24 +7,16 @@ import (
 
 	"github.com/openfga/openfga/pkg/id"
 	"github.com/openfga/openfga/pkg/logger"
-	"github.com/openfga/openfga/pkg/testutils"
 	"github.com/openfga/openfga/server/commands"
 	"github.com/openfga/openfga/server/errors"
 	"github.com/openfga/openfga/storage"
 	"github.com/stretchr/testify/require"
 	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
 )
 
-func TestWriteAuthorizationModel(t *testing.T, datastore storage.OpenFGADatastore) {
-	type writeAuthorizationModelTestSettings struct {
-		name    string
-		request *openfgapb.WriteAuthorizationModelRequest
-		err     error
-	}
-
-	ctx := context.Background()
-	logger := logger.NewNoopLogger()
+const storeID = "01GCD5SAJVC2Y38MJBNJ9QVZ60"
 
+func TestWriteAuthorizationModel(t *testing.T, datastore storage.OpenFGADatastore) {
```

Comment:
Can we add a test that uses TupleToUserset to verify the `tupleset` relation validates against a defined relation as well?

Reply:
Doesn't `succeeds part II` have such?

Reply:
Sorry, I put this on the wrong test. I meant for the tests with type restrictions.

I don't see a test that involves union, intersection, exclusion or tuple_to_userset with assignable vs non-assignable types.

Reply:
Ideally these kind of tests would belong on the helper methods in the `typesystem` package though like we discussed. That way it's more clear what paths of these critical helpers have been fully tested.